### PR TITLE
[4.0] Correcting massmail js

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/25509

### Summary of Changes
Corrected deprecated function and implemented joomla-alert

### Testing Instructions
Try to send a mail via massmail
1. Without subject
2. Without text

### After patch
Correct alerts.
No console error.

### CODE REVIEW

File: build/media_source/com_users/js/admin-users-mail.es6.js [hunk 1]
```diff
@@ -9,21 +9,23 @@
...
       }
 
       // do field validation
       if (form.jform_subject.value === '') {
-        // TODO replace with joomla-alert
-        alert(Joomla.JText._('COM_USERS_MAIL_PLEASE_FILL_IN_THE_SUBJECT'));
-      } else if (Joomla.getSelectedValue('adminForm', 'jform[group]') < 0) {
-        // TODO replace with joomla-alert
-        alert(Joomla.JText._('COM_USERS_MAIL_PLEASE_SELECT_A_GROUP'));
+        html.innerHTML = Joomla.JText._('COM_USERS_MAIL_PLEASE_FILL_IN_THE_SUBJECT');
```
Comment (C-Lodder) [inline build/media_source/com_users/js/admin-users-mail.es6.js]: Please replace `html.innerHTML` with `html.innerText`

File: build/media_source/com_users/js/admin-users-mail.es6.js [hunk 2]
```diff
@@ -9,21 +9,23 @@
...
+        form.insertAdjacentElement('afterbegin', html);
+      } else if (form.jform_group.value < 0) {
+        html.innerHTML = Joomla.JText._('COM_USERS_MAIL_PLEASE_SELECT_A_GROUP');
```
Comment (C-Lodder) [inline build/media_source/com_users/js/admin-users-mail.es6.js]: and here

File: build/media_source/com_users/js/admin-users-mail.es6.js [hunk 3]
```diff
@@ -9,21 +9,23 @@
...
+        form.insertAdjacentElement('afterbegin', html);
       } else if (form.jform_message.value === '') {
-        // TODO replace with joomla-alert
-        alert(Joomla.JText._('COM_USERS_MAIL_PLEASE_FILL_IN_THE_MESSAGE'));
+        html.innerHTML = Joomla.JText._('COM_USERS_MAIL_PLEASE_FILL_IN_THE_MESSAGE');
```
Comment (C-Lodder) [inline build/media_source/com_users/js/admin-users-mail.es6.js]: and here

**Review by C-Lodder — CHANGES_REQUESTED** [review]

**Comment by infograf768** [comment]
@C-Lodder
done

File: build/media_source/com_users/js/admin-users-mail.es6.js [hunk 4]
```diff
@@ -9,21 +9,23 @@
...
       // do field validation
       if (form.jform_subject.value === '') {
-        // TODO replace with joomla-alert
-        alert(Joomla.JText._('COM_USERS_MAIL_PLEASE_FILL_IN_THE_SUBJECT'));
-      } else if (Joomla.getSelectedValue('adminForm', 'jform[group]') < 0) {
-        // TODO replace with joomla-alert
-        alert(Joomla.JText._('COM_USERS_MAIL_PLEASE_SELECT_A_GROUP'));
+        html.innerText = Joomla.JText._('COM_USERS_MAIL_PLEASE_FILL_IN_THE_SUBJECT');
+        form.insertAdjacentElement('afterbegin', html);
+      } else if (form.jform_group.value < 0) {
```
Comment (wilsonge) [inline build/media_source/com_users/js/admin-users-mail.es6.js:23]: Shouldn't this be less than or equal to 0?

Comment (infograf768) [reply build/media_source/com_users/js/admin-users-mail.es6.js:23]: If it is equal to zero, we should not have any alert as 0 means All User Groups
```
		<field 
			name="group" 
			type="usergrouplist"
			label="COM_USERS_MAIL_FIELD_GROUP_LABEL"
			default="0"
			size="1"
			>
			<option value="0">COM_USERS_MAIL_FIELD_VALUE_ALL_USERS_GROUPS</option>
		</field>
```
btw, I wonder why we even have this alert at all as it looks impossible to get a negative value.
I see it is already present in 3.0, 2.5 and 1.6 although not in a separate js
```
$script .= "\t\t".'} else if (getSelectedValue(\'adminForm\',\'jform[group]\') < 0){'."\n";
$script .= "\t\t\t".'alert("'.JText::_('COM_USERS_MAIL_PLEASE_SELECT_A_GROUP', true).'");'."\n";
```

**Comment by infograf768** [comment]
Drone failure looks unrelated to this PR.

**Comment by wilsonge** [comment]
Thanks

**Pull request merged**